Permanent/App/AppDelegate.swift

```diff
@@ -84,15 +86,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             }
             
         case "app":
-            if url.pathComponents.count >= 3, url.pathComponents[2] == "pr" && url.pathComponents[3] == "manage" {
-                if rootViewController.isDrawerRootActive {
-                    return navigateFromSharedArchive()
+            if let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+               let redeemCode = components.queryItems?.first(where: { $0.name == "promoCode" })?.value {
+                if let authData = KeychainSwift().getData(SessionKeychainHandler.keychainAuthDataKey),
+                   let session = try? JSONDecoder().decode(PermSession.self, from: authData),
+                   let archiveId = session.selectedArchive?.archiveID,
+                   archiveId != .zero {
+                    let storageViewModel = StateObject(wrappedValue: StorageViewModel(reddemCode: redeemCode))
+                    let storageView = StorageView(viewModel: storageViewModel)
+                    
+                    let host = UIHostingController(rootView: storageView)
+                    self.window?.rootViewController?.present(host, animated: true, completion: nil)
+                    
+                    return true
                 } else {
-                    saveSharedArchiveToken()
                     return false
                 }
+            } else {
+                if url.pathComponents.count >= 3, url.pathComponents[2] == "pr" && url.pathComponents[3] == "manage" {
```

Comment:
I think it should be url.pathComponents.count > 3 as you are accessing the url.pathComponents[3]and this could lead to a crash.

Reply:
This was the previous implementation, should we look  into it?

Reply:
Resolved this issue with the last commit.